internal/worker_runner.go

```diff
@@ -62,37 +69,68 @@ func (wr *workerRunner) Run() {
 	wr.logger.Info().Msgf("found modem: %s", modem)
 
 	// we will need two clocks, one for non-obd (every 20s) and one for obd (continuous, based on each signal interval)
-	// which clock checks batteryvoltage? we want to send it with every status payload
-	// register tasks that can be iterated over
+	// battery-voltage will be checked in obd related clock to determine if it is ok to query obd
+	// battery-voltage also will be checked in non-obd clock because we want to send it with every status payload
+	go func() {
+		fingerprintDone := false
+		for {
+			// we will need to check the voltage before we query obd, and then we can query obd if voltage is ok
+			queryOBD, powerStatus := wr.isOkToQueryOBD()
+			if queryOBD {
+				fmt.Printf("voltage is enough to query obd : %f\n", powerStatus.VoltageFound)
+				// do fingerprint but only once
+				if !fingerprintDone {
+					err := wr.fingerprintRunner.FingerprintSimple(powerStatus)
+					if err != nil {
+						wr.logger.Err(err).Msg("failed to do vehicle fingerprint")
+					} else {
+						fingerprintDone = true
+					}
+				}
+				// query OBD signals
+				wr.queryOBD()
+			} else {
+				wr.logger.Info().Msg("voltage not enough to query obd")
+			}
+
+			time.Sleep(2 * time.Second)
+		}
+	}()
+
+	// Note: this delay required for the tests only, to make sure that queryOBD is executed before nonObd signals
+	time.Sleep(1 * time.Second)
 	for {
-		s := wr.createDeviceEvent(modem)
+		select {
+		case <-wr.stop:
+			// If stop signal is received, stop the loop
+			// Note: this is used only for unit/functional tests
+			fmt.Println("Stopping worker runner")
+			return
+		default:
+			_, powerStatus := wr.isOkToQueryOBD()
```

Comment:
@JamesReate , I added a voltage check in non-obd loop as well, since we want to include it in payload